diff --git a/index.js b/index.js
index e5cbb1718b06dc5124f7a58ecb4e2ea61c7133ce..a0c2f6c4ef03673ff88de34beb2518e83b515267 100644
--- a/index.js
+++ b/index.js
@@ -521,9 +521,9 @@ class CLICanvas {
         textureBuffer.unmap();
         return;
       }
-      const mappedRangePtr = textureBuffer.getMappedRangePtr(0, textureBuffer.size);
-      const bufPtr = mappedRangePtr;
       if (this.superSample === "cpu" /* CPU */) {
+        const mappedRangePtr = textureBuffer.getMappedRangePtr(0, textureBuffer.size);
+        const bufPtr = mappedRangePtr;
         const format = contextFormat === "bgra8unorm" ? "bgra8unorm" : "rgba8unorm";
         const ssStart = performance.now();
         buffer.drawSuperSampleBuffer(0, 0, bufPtr, textureBuffer.size, format, alignedBytesPerRow);
